### PR TITLE
fix - query by date

### PR DESCRIPTION
**Pull request by mdayican**
### Change description ###

fix - query by date in `SupplierStatementRepository`

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/statements/SupplierStatementRepository.java [hunk 1]
```diff
@@ -64,10 +64,10 @@ public UUID save(NewEnvelopeSupplierStatement statement) throws SQLException {
         }
     }
 
-    public Optional<EnvelopeSupplierStatement> findLatest(LocalDate date) {
+    public Optional<EnvelopeSupplierStatement> findLatestByDate(LocalDate date) {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/statements/SupplierStatementRepository.java:67]: Now this needs javadoc for sure :)

File: src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/statements/SupplierStatementRepository.java [hunk 2]
```diff
@@ -64,10 +64,10 @@ public UUID save(NewEnvelopeSupplierStatement statement) throws SQLException {
...
         try {
             EnvelopeSupplierStatement statement = jdbcTemplate.queryForObject(
-                "SELECT * FROM envelope_supplier_statements WHERE DATE(created_at) = :date "
+                "SELECT * FROM envelope_supplier_statements WHERE date = :date "
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/statements/SupplierStatementRepository.java:70]: The query changed completely, but tests did not. Looks like we need an extra test?

**Review by rkondratowicz — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/statements/SupplierStatementRepository.java (hunk 1 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/statements/SupplierStatementRepository.java]: changed the method name.

File: src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/statements/SupplierStatementRepository.java (hunk 2 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/statements/SupplierStatementRepository.java:70]: change the date in test, it ws using the same date.

**Pull request closed without merging**